runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningRocksOverlay.java

```diff
@@ -54,25 +60,39 @@ class MiningRocksOverlay extends Overlay
 
 	private final Client client;
 	private final MiningPlugin plugin;
+	private final MiningConfig config;
 
 	@Inject
-	private MiningRocksOverlay(Client client, MiningPlugin plugin)
+	private MiningRocksOverlay(Client client, MiningConfig config, MiningPlugin plugin)
 	{
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_SCENE);
 		this.plugin = plugin;
+		this.config = config;
 		this.client = client;
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
 		List<RockRespawn> respawns = plugin.getRespawns();
-		if (respawns.isEmpty())
+		Map<TileObject, Tile> camTorumStreams = plugin.getCamTorumStreams();
+
+		if (!respawns.isEmpty())
```

Comment:
I don't think either of these empty checks are necessary as the only thing the `handle` methods do are loops over these collections, which is effectively free for empty collections.